src/Fusonic/Linq/Linq.php

```diff
@@ -120,7 +120,8 @@ public function skip($count)
             }
         }
 
-        return new Linq(new \LimitIterator($innerIterator, $count, -1));
+        // IteratorIterator wraps $innerIterator because it might be Traversable but not an Iterator.
+        return new Linq(new \LimitIterator(new \IteratorIterator($innerIterator), $count, -1));
```

Comment:
It might be worth checking for the $innerIterator type and only wrap it if`s necessary. This would save an unneeded allocation + indirection during execution.